controlplane/eks/api/v1beta1/zz_generated.conversion.go

```diff
@@ -360,6 +360,7 @@ func autoConvert_v1beta1_AWSManagedControlPlaneSpec_To_v1beta2_AWSManagedControl
 	if err := Convert_v1beta1_KubeProxy_To_v1beta2_KubeProxy(&in.KubeProxy, &out.KubeProxy, s); err != nil {
 		return err
 	}
+	out.RestrictPrivateSubnets = in.RestrictPrivateSubnets
```

Comment:
Look at how the `vpccni.Disable` field works for conversion.
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controlplane/eks/api/v1beta1/conversion.go#L29-L61

I think what's missing for the fuzzy tests to pass (without the v1beta1 field) is that when downgrading to v1beta1, we need to store the lost field state in an annotation value, and then we upgrading, we need to extract the state back out and restore the lost fields. There's already a lot of machinery for this boilerplate built, that is what the `utilconversion` package is for. `utilconversion.MarshalData` will store state into an annotation and `utilconversion.UnmarshalData` will extract the state back out. Take a look at some other examples of `utilconversion` to get a feel for it.

Reply:
Thank you for the review and the pointer. I did have a few unsuccessful attempts with `utilconversion` to restore the field. I'll review the example you mentioned.

Reply:
It's certainly not intuitive and I had to have another maintainer walk me through it the first time. If you get stuck, we're also around on the kubernetes slack, which may be faster for iteration.

Reply:
looks like it just needed to be restored during the v1 to v2 conversion. CI is passing now.